packages/react-components/react-table/Spec.md

```diff
@@ -2,62 +2,547 @@
+  // sort data
+};
+
+const columns: IColumn[] = [
+  {
+    name: 'File Type',
+    fieldName: 'name',
+    onColumnClick: onColumnClick,
+  },
+  {
+    name: 'Name',
+    fieldName: 'name',
+    onColumnClick: onColumnClick,
+  },
+];
+
+<DetailsList
+  items={items}
+  columns={columns}
+  isHeaderVisible={true}
+/>
+```
+
+#### Column resizing
+
+Column resizing is only supported by the v8 DetailsList. Columns can be resized without any additional configuration.
+In order to support this, every column always has an inline style where the width in pixels is applied.
 
 ## Sample Code
 
-_Provide some representative example code that uses the proposed API for the component_
+```tsx
+<Table>
+  <TableHeader>
+    <TableRow>
+      <TableCell> </TableCell>
+      <TableCell> </TableCell>
+      <TableCell> </TableCell>
+    </TableRow>
+  </TableHeader>
+
+  <TableBody>
+    <TableRow>
+      <TableCell> </TableCell>
+      <TableCell> </TableCell>
+      <TableCell> </TableCell>
+    </TableRow>
+  </TableBody>
+</Table>
+```
 
 ## Variants
 
-_Describe visual or functional variants of this control, if applicable. For example, a slider could have a 2D variant._
+### Sorting
+
+The table header should support design and interactions for sorting the table by column.
+
+### Selectable rows
+
+Table rows can be selected. The table header should also support a 'select all' feature, selecting the
+table header should select all rows.
+
+### Primary column
+
+A primary column is generally the first column of the table, however there is no strict requirement on this. This
+column has some design differences and supports secondary content which can contain extra instructions or
+description.
+
+### Sizes
+
+The table supports the following sizes that affect the layout and size of its child components:
+
+- small
+- smaller
+- medium
+- large
 
 ## API
 
-_List the **Props** and **Slots** proposed for the component. Ideally this would just be a link to the component's `.types.ts` file_
+> Current proposed API, will be eventually be replaced with a hard link to the `Component.types.ts` file
+> more mature into development.
+
+### Table
+
+The `Table` component is intended to present data in a tabular format. Apart from sortable headers, the component
+is intended to be presentational and not interactive. This component can also be a bail out for end users if
+overriding the default interaction behaviour of the `DataGrid` component is too difficult.
+
+```ts
+interface TableProps {
+  root: Slot<'table' | 'div'>;
+  /**
+   * Determines sizes of child components
+   * @default medium
+   */
+  size: 'small' | 'smaller' | 'medium' | 'large';
+  /**
+   * Whether the columns are sortable
+   */
+  sortable?: boolean;
+
+  /**
+   * Renders all table components as `div` with appropriate roles.
+   * Avoids manual `as` overrides for all table components
+   *
+   * @default true
+   */
+  nativeElements?: boolean;
```

Comment:
Perhaps we could simply use `slot` and if `div` is provided you can assume `div` with appropriate role for the rest of the compound components. 

Or...How about using a default `false` property?

Reply:
I do want automatic propagation of `div`to the compound components but I would rather not do it at the expense of API consistency, this would be the first example of slot propagation and I don't know if it's worth introducing since I can't think of any other component that would need it.

I don't mind doing a default `false` property... I just have to find an appropriate name for it 😄

Reply:
`noNativeElement` 🤣